specs/pvs/corla.pvs

```diff
@@ -1420,8 +1457,71 @@ BEGIN
   number_of_ballots_to_audit: [audited_contest -> nat]
 
 %%% \todokiniry{Still need to model core RLA algorithm(s).}
+
+% @trace RDPR-6-Jul-2017 Section 25.1.4
+% @description A type of risk-limiting audit in which the audit board
+% examines and reports to the secretary of state voter markings on
+% randomly selected ballots, and then compared compares them to the
+% voting system's tabulation as reflected in the corresponding cast
+% vote records.
+  comparison_audit: TYPE FROM risk_limiting_audit
 END rlas
 
+%%% In order to instantiate the general idea of Risk-Limiting Audits
+%%% into the context of Colorado elections, we must: (1) choose RLA
+%%% parameters appropriate for Colorado's context, and (2) instantiate
+%%% the operation of the concurrent RLA across all counties in a
+%%% fashion that fits the needs of the CDOS.
+
+colorado_rlas: THEORY
+BEGIN
+  IMPORTING rlas
+
+%%% The total number of ballot cards uploaded across the entire state.
+  total_ballot_cards: nat
+
+%%% Next, we'll formalize what is stated in the current draft
+%%% rules. First, the risk limit for comparison audits must be below
+%%% $5\%$.
+% @trace RDPR-6-Jul-2017 Section 25.2.2(A)
+  RDPR_risk_limit: TYPE = {n : nonneg_real | n <= 0.05}
+
+%%% Also, according to the current draft rules, the Secretary of State
+%%% has an ``escape clause'', and one thing that they can choose to do
+%%% is set any kind of risk limit they like.
+  RDPR_escape_clause_risk_limit: TYPE = {n : nonneg_real | n <= 1.0}
+
+%%% As mentioned above, CDOS wants to be able to run a state-wide
+%%% audit in a number of ``rounds'', or what are known as ``stages''
+%%% in the RLA literature. Thus, rounds need to be enumerated with an
+%%% ordinal, such as ```We are now starting round 3 of the Colorado
+%%% Risk-Limiting Audit.''.
+  round_number: TYPE = posnat
+%%% In each round, a specific number of ballots will be audited by
+%%% counties.
+  round_size: TYPE = [round_number -> nat]
+%%% Eventually we run out of ballot cards to audit, as a full hand
+%%% count will count all ballot cards. Thus, for all possible round
```

Comment:
I'm proposing a model of stages such that stages eventually terminate because the SOS will decide that the contest(s) in question must be hand-counted.  I'll try to clarify.